### PR TITLE
Proposed update of Community Installation Policy 10

### PR DESCRIPTION
**Pull request by balos1**
Continuing from https://github.com/xsdk-project/xsdk-issues/issues/79.

CC: @balay @mkstoyanov @bartlettroscoe

### CODE REVIEW

**Review by bartlettroscoe — APPROVED** [review]
The changes look good to me, at least an improvement.

File: installation_policies/discussion_and_examples.md [hunk 1]
```diff
@@ -47,21 +47,17 @@ For configure we are trying to match as closely as possible the GNU autoconf and
     If the user does provide one, however, it **must** be used; if it is not able to be used, then an error must be 
     generated. A package cannot silently substitute a different installation.
 
-10. `./configure --with-x --with-metis-lib=/usr/local/lib/libmetis.a --with-metis-include=-I/usr/local/include  ; cmake -DTPL_Metis_LIBRARIES=/usr/local/lib/libmetis.a -DTPL_Metis_INCLUDE_DIRS=/usr/local/include ...`
+10.  `./configure --with-x --with-metis-lib=/usr/local/lib/libmetis.a --with-metis-include=-I/usr/local/include`
```
Comment (balay) [inline installation_policies/discussion_and_examples.md]: Should be:
`--with-metis-include=/usr/local/include`

File: installation_policies/discussion_and_examples.md [hunk 2]
```diff
@@ -47,21 +47,17 @@ For configure we are trying to match as closely as possible the GNU autoconf and
...
-    * We also do not recommend support --with-<package>-dir=directory because that is difficult to implement robustly.
-
-    * Packages are free to locate a package on the file system if none is specifically provided by the user. If the user does 
-    provide one, however, it **must **be used; if it is not able to be used, then an error must be generated. A package 
-    cannot silently substitute a different installation.
-
-    * There does not exist any CMake standard allowing an external user to set what external package dependencies should be 
-    enabled or disabled when configuring.   Therefore, this is a TriBITS/Trilinos standard is which calls external packages 
-    "TPL"s and therefore the name “TPL_ENABLE_<package>”.
+     * In CMake, the analogous approach would be `cmake -DTPL_ENABLE_METIS=ON -DTPL_METIS_INCLUDE_DIRS=-I/usr/local/include -DTPL_METIS_LIBRARIES=/usr/local/lib/libmetis.a`. However, a package may use CMake's `find_package()` command to load a dependent library as long as the package provides a way for a user to specify an installation of the dependent library to use, *and* the package guarantees that the specified installation is not substituted.
```
Comment (balay) [inline installation_policies/discussion_and_examples.md]: Should be:
`-DTPL_METIS_INCLUDE_DIRS=/usr/local/include`

File: installation_policies/10.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
 10. Determine other package libraries and include directories.
 
-    1. `--with-<package>  --with-<package>-lib="linkable list of libraries” --with-package-include=”-I list of include
```
Comment (balay) [inline installation_policies/10.md:3]: I'm not sure if we should be changing this autoconf part - i.e retain the --with-package-include and --with-pacakge-lib options?

File: installation_policies/discussion_and_examples.md (hunk 2 above)
Comment (bartlettroscoe) [reply installation_policies/discussion_and_examples.md]: Yes!

File: installation_policies/10.md (hunk 3 above)
Comment (balos1) [reply installation_policies/10.md:3]: I am not too familiar with autoconf and this policy has not been an issue with any packages that use autoconf, so we probably should retain it. I will add it back in unless anyone else has objections.

**Comment by balos1** [comment]
This was approved by a vote with no objections, today (5/16/19 11:50 AM), on the xSDK meeting call.

File: installation_policies/10.md (hunk 3 above)
Comment (balos1) [reply installation_policies/10.md:3]: The vote today included retaining this line in.

**Pull request merged**